server/src/test/java/io/crate/expression/scalar/SubscriptFunctionTest.java

```diff
@@ -43,6 +43,22 @@ public void test_long_can_be_used_as_array_index() {
         assertEvaluate("['Youri', 'Ruben'][x]", "Youri", Literal.of(1L));
     }
 
+    @Test
+    public void test_index_out_of_range_array_access() {
+        assertThatThrownBy(
+            () -> assertEvaluate("['Youri', 'Ruben'][x]",
+                                 null,
+                                 Literal.of((long) Integer.MAX_VALUE + 1)))
+            .isExactlyInstanceOf(ConversionException.class)
+            .hasMessage("Cannot cast value `2147483648` to type `integer`");
```

Comment:
Questions: where does this cast error is coming from since you removed it from `SubscriptValidator`?

Reply:
As part of allocating functions (`SubscriptFunction` in this case) `ExpressionAnalzer.cast` converts the arguments to the target types.